projects/sonar/src/app/record/files/stats-files/stats-files.component.ts

```diff
@@ -47,28 +88,36 @@ export class StatsFilesComponent implements OnInit {
         stat: 'record-view',
         params: {
           pid_value: this.record().pid,
-          pid_type: 'doc'
-        }
+          pid_type: 'doc',
+        },
       },
       'file-download': {
         stat: 'file-download',
         params: {
-          bucket_id: this.record()._bucket
-        }
-      }
+          bucket_id: this.record()._bucket,
+        },
+      },
     };
 
-    this.httpClient.post(`${this.apiService.getEndpointByType('stats', true)}`, data)
-    .subscribe(results => {
-      const statistics = {};
-      if (results['file-download'] != null) {
-        results['file-download'].buckets.map(
-          res => statistics[res.key] = res.unique_count
-        );
-      }
-      statistics['record-view'] = results['record-view'].unique_count;
-      this.statistics.set(statistics);
-    });
+    this.httpClient
+      .post(`${this.apiService.getEndpointByType('stats', true)}`, data)
+      .subscribe((results) => {
+        const statistics: StatisticsData = {
+          'record-view': { count: 0 },
+          'file-download': {},
+        };
+        if (results['file-download'] != null) {
```

Comment:
`if (results['file-download'] !== undefined)` would be cleaner? Or simply `if (results['file-download'])` ?